### PR TITLE
New feature

### PR DESCRIPTION
**Pull request by MStraczkiewicz**
Please review and merge changes to the code. The changes regard shortening computational time required to process Oak using Beiwe data.

### CODE REVIEW

File: forest/oak/base.py [hunk 1]
```diff
@@ -592,7 +624,7 @@ def run(study_folder: str, output_folder: str, tz_str: str = None,
                     bout_time = [t_i.to_pydatetime() for t_i in bout_time[:-1]]
 
                     # find observations in this bout
-                    acc_ind = np.isin(t_shifted, bout_time)
+                    acc_ind = ismember(t_shifted, bout_time)
```
Comment (hackdna) [inline forest/oak/base.py:627]: How about something like:
```
acc_ind = [observation in bout_time for observation in t_shifted]
```
Also, it'd be great to give `acc_ind` a less cryptic name.

**Review by hackdna — CHANGES_REQUESTED** [review]

File: forest/oak/base.py [hunk 2]
```diff
@@ -532,128 +523,98 @@ def run(study_folder: str, output_folder: str, tz_str: str = None,
...
-                                                  z_bout)[3]
-
-                        # find walking and estimate steps
-                        cadence_bout = find_walking(vm_bout)
-                        cadence_bout = cadence_bout[np.where(cadence_bout
-                                                             > 0)]
-                        cadence_temp_daily.append(cadence_bout)
-                        cadence_temp_hourly.append(cadence_bout)
-
+            data = pd.DataFrame()
```
Comment (hackdna) [inline forest/oak/base.py:526]: Python does not require initializing variables, so this is unnecessary here.

File: forest/oak/base.py [hunk 3]
```diff
@@ -461,6 +439,7 @@ def run(study_folder: str, output_folder: str, tz_str: str = None,
         fs: integer
             sampling frequency
```
Comment (hackdna) [inline forest/oak/base.py:440]: Parameter `fs` value is no longer used, so should be removed from function signature and docstring

File: forest/oak/base.py [hunk 4]
```diff
@@ -532,128 +523,98 @@ def run(study_folder: str, output_folder: str, tz_str: str = None,
...
-                        cadence_bout = find_walking(vm_bout)
-                        cadence_bout = cadence_bout[np.where(cadence_bout
-                                                             > 0)]
-                        cadence_temp_daily.append(cadence_bout)
-                        cadence_temp_hourly.append(cadence_bout)
-
+            data = pd.DataFrame()
+
+            # load data for a given day
+            if len(file_ind) > 0:
```
Comment (hackdna) [inline forest/oak/base.py:529]:
```
if len(file_ind) <= 0:
    continue
```
It's easier to read and saves one level of indentation for the remaining code.

File: forest/oak/base.py [hunk 5]
```diff
@@ -532,128 +523,98 @@ def run(study_folder: str, output_folder: str, tz_str: str = None,
...
+                x = np.array(data["x"], dtype="float64")  # x-axis acc.
+                y = np.array(data["y"], dtype="float64")  # y-axis acc.
+                z = np.array(data["z"], dtype="float64")  # z-axis acc.
+
+                # preprocess data fragment
+                t_bout_interp, vm_bout = preprocess_bout(timestamp, x, y, z)
+
+                # get t as datetimes
+                t_datetime = [datetime.fromtimestamp(t_ind) for t_ind in
+                              t_bout_interp]
```
Comment (hackdna) [inline forest/oak/base.py:549]: A bit easier to read:
```
                t_datetime = [datetime.fromtimestamp(t_ind)
                              for t_ind in t_bout_interp]
```

**Review by hackdna — CHANGES_REQUESTED** [review]

File: forest/oak/base.py (hunk 2 above)
Comment (MStraczkiewicz) [reply forest/oak/base.py:526]: I initiate `data` for `data = pd.concat([data, pd.read_csv(file_path)], axis=0)`. Is there an easier way to go?

**Review by hackdna — CHANGES_REQUESTED** [review]
Thanks!

**Review by hackdna — APPROVED** [review]

**Comment by hackdna** [comment]
Thank you. I will be able to merge once you resolve the conflicts with the develop branch.

**Review by hackdna — APPROVED** [review]

**Pull request merged**